File: leetcode/0283-move-zeroes/moveZeroes.test.ts
import cases from "./cases.json";
import missingNumber from "./moveZeroes";

test("move zeroes", () => {
  cases.forEach((testCase, index) => {
    console.log(`正在执行第 ${index + 1} 个测试用例\n`);
    const result = missingNumber(testCase.input);
    console.log(
      `input: ${testCase.input} , output: ${testCase.output} ==> ${result} \n`
    );

    expect(testCase.input).toEqual(testCase.output);
  });
});

File: leetcode/0283-move-zeroes/moveZeroes_test.go
package leetcode

import (
	"fmt"
	"path/filepath"
	"reflect"
	"testing"

	"github.com/edram/leetcode/util"
)

type MoveZeroesCase struct {
	Input []int

	Output []int
}

func TestMoveZeroes(t *testing.T) {
	var casePath, _ = filepath.Abs("./cases.json")
	payload, err := util.LoadCases[[]MoveZeroesCase](casePath)
	if err != nil {
		panic(err)
	}

	for i, testCase := range *payload {
		fmt.Printf("正在执行第 %d 个测试用例\n", i+1)
		fmt.Printf("string: %v ==> %v \n", testCase.Input, testCase.Output)

		moveZeroes(testCase.Input)
		if !reflect.DeepEqual(testCase.Input, testCase.Output) {
			t.Errorf("期望 %v, 结果 %v", testCase.Output, testCase.Input)
		}
	}
}
